Sources/PointFree/Account/UpdateProfile.swift

```diff
@@ -22,24 +24,43 @@ extension ProfileData: Decodable {
     let container = try decoder.container(keyedBy: CodingKeys.self)
 
     self.email = try container.decode(EmailAddress.self, forKey: .email)
-    self.name = try container.decodeIfPresent(String.self, forKey: .name)
     self.emailSettings = (try? container.decode([String: String].self, forKey: .emailSettings)) ?? [:]
+    self.extraInvoiceInfo = try? container.decode(String.self, forKey: .extraInvoiceInfo)
+    self.name = try container.decodeIfPresent(String.self, forKey: .name)
   }
 }
 
 func isValidEmail(_ email: EmailAddress) -> Bool {
   return email.rawValue.range(of: "^.+@.+$", options: .regularExpression) != nil
 }
 
+private func fetchStripeSubscription<A>(
+  _ middleware: (@escaping Middleware<StatusLineOpen, ResponseEnded, T2<Stripe.Subscription?, A>, Data>)
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.Subscription?, A>, Data> {
+
+    return { conn -> IO<Conn<ResponseEnded, Data>> in
+      guard let subscription = get1(conn.data)
+        else { return middleware(conn.map(over1(const(Stripe.Subscription?.none)))) }
+
+      return Current.stripe.fetchSubscription(subscription.stripeSubscriptionId)
+        .run
+        .map(^\.right)
+        .flatMap { conn.map(const($0 .*. conn.data.second)) |> middleware }
+    }
+}
+
 let updateProfileMiddleware =
-  filterMap(require2 >>> pure, or: loginAndRedirect)
-    <<< filterMap(require1 >>> pure, or: redirect(to: .account(.index)))
```

Comment:
I changed the order of the connection's data to include the user first and then the `ProfileData`, so some `1`s and `2`s got flipped.